components/update_user_group_modal/update_user_group_modal.tsx

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useState} from 'react';
+
+import {Modal} from 'react-bootstrap';
+
+import {FormattedMessage} from 'react-intl';
+
+import * as Utils from 'utils/utils.jsx';
+import {CustomGroupPatch, Group} from 'mattermost-redux/types/groups';
+
+import 'components/user_groups_modal/user_groups_modal.scss';
+import './update_user_group_modal.scss';
+import {ModalData} from 'types/actions';
+import Input from 'components/input';
+import {ActionResult} from 'mattermost-redux/types/actions';
+import LocalizedIcon from 'components/localized_icon';
+import {t} from 'utils/i18n';
+
+import SaveButton from 'components/save_button';
+
+export type Props = {
+    onExited: () => void;
+    groupId: string;
+    group: Group;
+    backButtonCallback: () => void;
+    actions: {
+        patchGroup: (groupId: string, group: CustomGroupPatch) => Promise<ActionResult>;
+        openModal: <P>(modalData: ModalData<P>) => void;
+    };
+}
+
+const UpdateUserGroupModal = (props: Props) => {
+    const [hasUpdated, setHasUpdated] = useState(false);
+    const [name, setName] = useState(props.group.display_name);
+    const [mention, setMention] = useState(`@${props.group.name}`);
+    const [saving, setSaving] = useState(false);
+    const [show, setShow] = useState(true);
+    const [mentionInputErrorText, setMentionInputErrorText] = useState('');
+    const [nameInputErrorText, setNameInputErrorText] = useState('');
+    const [showUnknownError, setShowUnknownError] = useState(false);
+    const [mentionUpdatedManually, setMentionUpdatedManually] = useState(false);
+
+    const doHide = () => {
+        setShow(false);
+    };
+
+    const isSaveEnabled = () => {
+        return name.length > 0 && mention.length > 0 && hasUpdated && !saving;
+    };
+
+    const updateNameState = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const value = e.target.value;
+        let newMention = mention;
+        if (!mentionUpdatedManually) {
+            newMention = value.replace(/[^A-Za-z0-9@]/g, '').toLowerCase();
+            if (newMention.substring(0, 1) !== '@') {
+                newMention = `@${newMention}`;
+            }
+        }
+        setName(value);
+        setHasUpdated(true);
+        setMention(newMention);
+    };
+
+    const updateMentionState = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const value = e.target.value;
+        setHasUpdated(true);
+        setMention(value);
+        setMentionUpdatedManually(true);
+    };
+
+    const patchGroup = async () => {
```

Comment:
Again, all this callback functions could take advantage of something like `useCallback`